src/docfx/serve/lsp/TextDocumentHandler.cs

```diff
@@ -16,20 +17,42 @@ namespace Microsoft.Docs.Build
 {
     internal class TextDocumentHandler : ITextDocumentSyncHandler
     {
-        private readonly Channel<FileActionEvent> _channel;
-        private readonly DocumentSelector _documentSelector = new(new DocumentFilter { Pattern = "**/*.{md,yml,json}" });
+        private static readonly string[] s_configFileNames =
+            {
+                "docfx.json",
+                "docfx.yml",
+                ".openpublishing.publish.config.json",
+                "docsets.json",
+                "docsets.yml",
+                "redirections.yml",
+                "redirections.json",
+                ".openpublishing.redirection.json",
+            };
+
+        private readonly SemaphoreSlim _buildSemaphore;
+        private readonly LanguageServerPackage _package;
+
+        private readonly DocumentSelector _documentSelector = new(
+            new DocumentFilter()
+            {
+                Pattern = "**/*.{md,yml,json}",
+            });
 
-        public TextDocumentHandler(Channel<FileActionEvent> channel)
+        public TextDocumentHandler(SemaphoreSlim buildSemaphore, LanguageServerPackage package)
```

Comment:
Encapsulate `SemaphoreSlim` as an implementation detail, prefer a method like `QueueBuild`.

```suggestion
        public TextDocumentHandler(LanguageServerBuilder, LanguageServerPackage package)
```